ndcube/ndcube.py

```diff
@@ -964,24 +966,98 @@ def _new_instance(self, **kwargs):
     def __neg__(self):
         return self._new_instance(data=-self.data)
 
-    def __add__(self, value):
-        if hasattr(value, 'unit'):
+    def add(self, value, handle_mask=np.logical_and):
+        """
+        Users are allowed to choose hether they want handle_mask to be AND / OR .
+        """
+        kwargs = {}
+
+        if isinstance(value, NDData) and value.wcs is None:
+            if self.unit is not None and value.unit is not None:
+                value_data = (value.data * value.unit).to_value(self.unit)
+            elif self.unit is None and value.unit is None:
+                value_data = value.data
+            else:
+                raise TypeError("Adding objects requires both have a unit or neither has a unit.") # change the test as well.
+
+            # addition
+            kwargs["data"] = self.data + value_data
+            result_data = kwargs["data"]
+
+            # check whether there is a mask.
+            # Neither self nor value has a mask
+            self_unmasked = self.mask is None or self.mask is False or not self.mask.any()
+            value_unmasked = value.mask is None or value.mask is False or not value.mask.any()
+
+            if (self_unmasked and value_unmasked):
+                # combine the uncertainty, it can be propagated without any issue.
+                kwargs["uncertainty"] = self.combine_uncertainty(value, result_data)
+
+            elif (self_unmasked and not value_unmasked):
+                kwargs["mask"] = value.mask # mask needs to be set.
+                # combine the uncertainty
+                kwargs["uncertainty"] = self.combine_uncertainty(value, result_data)
+
+            elif (value_unmasked and not self_unmasked):
+                kwargs["mask"] = self.mask
+                # combine the uncertainty
+                kwargs["uncertainty"] = self.combine_uncertainty(value, result_data)
+
+            else:
+                kwargs["mask"] = handle_mask(self.mask,value.mask)
+                # combine the uncertainty
+                kwargs["uncertainty"] = self.combine_uncertainty(value, result_data)
+
```

Comment:
This can all be simplified to:
```suggestion
            kwargs["uncertainty"] = self._combine_uncertainty(value, result_data)
            # Handle mask.
            self_unmasked = self.mask is None or self.mask is False or not self.mask.any()
            value_unmasked = value.mask is None or value.mask is False or not value.mask.any()
            if self_unmasked and value_unmasked:
                pass
            elif value_unmasked:
                kwargs["mask"] = value.mask
            elif self_unmasked:
                kwargs["mask"] = self.mask
            else:
                kwargs["mask"] = handle_mask(self.mask, value.mask)

```
See comment below for reason for changing `combine_uncertainty` to `_combine_uncertainty`.